go/libraries/doltcore/sqle/dtables/unscoped_diff_table.go

```diff
@@ -74,18 +80,101 @@ func (dt *UnscopedDiffTable) Schema() sql.Schema {
 	}
 }
 
-// Partitions is a sql.Table interface function that returns a partition of the data. Currently data is unpartitioned.
-func (dt *UnscopedDiffTable) Partitions(*sql.Context) (sql.PartitionIter, error) {
-	return index.SinglePartitionIterFromNomsMap(nil), nil
+// Partitions is a sql.Table interface function that returns a partition of the data. Returns one
+// partition for working set changes and one partition for all commit history.
+func (dt *UnscopedDiffTable) Partitions(ctx *sql.Context) (sql.PartitionIter, error) {
+	return NewSliceOfPartitionsItr([]sql.Partition{
+		newDoltDiffPartition(workingSetPartitionKey),
+		newDoltDiffPartition(commitHistoryPartitionKey),
+	}), nil
 }
 
 // PartitionRows is a sql.Table interface function that gets a row iterator for a partition.
-func (dt *UnscopedDiffTable) PartitionRows(ctx *sql.Context, _ sql.Partition) (sql.RowIter, error) {
-	return NewUnscopedDiffTableItr(ctx, dt.ddb, dt.head)
+func (dt *UnscopedDiffTable) PartitionRows(ctx *sql.Context, partition sql.Partition) (sql.RowIter, error) {
+	if bytes.Equal(partition.Key(), workingSetPartitionKey) {
+		return dt.newWorkingSetRowItr(ctx)
+	} else if bytes.Equal(partition.Key(), commitHistoryPartitionKey) {
+		return dt.newCommitHistoryRowItr(ctx)
+	} else {
+		return nil, fmt.Errorf("unexpected partition: %v", partition)
+	}
+}
+
+func (dt *UnscopedDiffTable) newWorkingSetRowItr(ctx *sql.Context) (sql.RowIter, error) {
+	sess := dsess.DSessFromSess(ctx.Session)
+	rootsProvider := dsess.NewSessionStateAdapter(sess.Session, ctx.GetCurrentDatabase(), map[string]env.Remote{}, map[string]env.BranchConfig{}, map[string]env.Remote{})
```

Comment:
Don't think you need to go to all this trouble, just call `sess.GetRoots(dbName)`

Reply:
ooooh! that's **soo** much better! thanks!